regex-assembly/941210.ra

```diff
@@ -0,0 +1,183 @@
+##! Please refer to the documentation at
+##! https://coreruleset.org/docs/development/regex_assembly/.
+
+##!+ i
+
+##! This rule tries to match all the possible ways to write 'javascript' using
+##! unicode characters, html entities, and javascript escape sequences.
+
+##! Matched order is sequential: we first match 'j', then 'a', then 'v', etc.
+
+##!> define html_entity_prefix &#x?0*
+##!> define whitespaces \t|\n|\r
+
+##! html_encoded_whitespace is a list of all the possible ways to write an encoded whitespace
+##!> assemble
+  {{whitespaces}}
+  ##!> assemble
+    {{html_entity_prefix}}9;?
+    {{html_entity_prefix}}13;?
+    {{html_entity_prefix}}10;?
+    {{html_entity_prefix}}A;?
+    {{html_entity_prefix}}D;?
+    &tab;
+    &newline;
+    ##!=>
+  ##!<
+  ##!=>
+  *
+  ##!=< html_encoded_whitespace
+##!<
+
+##! all the possible ways to end the word 'javascript', plus whitespaces
+##!> assemble
+  :
+  ##!> assemble
```

Comment:
I don't think this is the spirit of the rule. The idea is to have all variations, right? The first one is always the canonical form, and then you have all the entities.

Reply:
True. It just makes it a little harder to read, as it looks like it would produce `:(?:...)`, when it actually produces `:|...`. IMO, you don't need the extra `assemble`. It's fine to have the canonical form as the first one. You could also just add comments, e.g.,:

```
  ##! canonical form
  :
  ##! alternative forms
  ...
```

Reply:
Agreed. Let's use that.